### PR TITLE
Example of property-based access control using temporal value

### PR DESCRIPTION
**Pull request by HannesSandberg**
cypher manual [PR](https://github.com/neo4j/docs-cypher/pull/1070)

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1887.surge.sh

Updated pages:
- [Property-based access control](https://neo4j-docs-operations-1887.surge.sh/operations-manual/5/authentication-authorization/property-based-access-control)

File: modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc [hunk 1]
```diff
@@ -97,4 +97,18 @@ The role `regularUsers` does not need to have `READ` privilege for the property
...
+----
+
+.Granting permission to `READ` all properties on nodes where the property `createdAt` is later than current date to role `regularUsers`:
+[source, syntax, role="noheader"]
+----
+GRANT READ {*} ON GRAPH * FOR (n) WHERE n.createdAt > date() TO regularUsers
+----
+[NOTE]
+====
+The `date()` function will be evaluated and the value used to evaluate the privilege will be the date at the time the property-based privilege was created.
```
Comment (phil198) [inline modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc:109]:
```suggestion
The `date()` function will be evaluated and the value used to evaluate the privilege will be the date at the time the property-based privilege was created. Bear this in mind when designing your property rules and make use of the `SHOW PRIVILEGES AS COMMANDS` command to check the stored value as it was evaluated. This is essential when revoking property-based privileges containing evaluated function values. 
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1887.surge.sh

Updated pages:
- [Property-based access control](https://neo4j-docs-operations-1887.surge.sh/operations-manual/5/authentication-authorization/property-based-access-control)

File: modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc [hunk 2]
```diff
@@ -97,4 +97,18 @@ The role `regularUsers` does not need to have `READ` privilege for the property
...
+----
+
+.Granting permission to `READ` all properties on nodes where the property `createdAt` is later than current date to role `regularUsers`:
+[source, syntax, role="noheader"]
+----
+GRANT READ {*} ON GRAPH * FOR (n) WHERE n.createdAt > date() TO regularUsers
+----
+[NOTE]
+====
+The `date()` function will be evaluated and the value used to evaluate the privilege will be the date at the time the property-based privilege was created. Bear this in mind when designing your property rules and make use of the `SHOW PRIVILEGES AS COMMANDS` command to check what the stored value is. This is essential when revoking property-based privileges containing evaluated function values like `date()`.
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc:109]: How about this wording:
```suggestion
The `date()` function is evaluated, and the value used to evaluate the privilege is the date when the property-based privilege is created. 
Keep this in mind when designing your property rules, and use the `SHOW PRIVILEGES AS COMMANDS` command to check the stored value.
This is essential when revoking property-based privileges containing evaluated function values like `date()`.
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1887.surge.sh

Updated pages:
- [Property-based access control](https://neo4j-docs-operations-1887.surge.sh/operations-manual/5/authentication-authorization/property-based-access-control)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1887.surge.sh

Updated pages:
- [Property-based access control](https://neo4j-docs-operations-1887.surge.sh/operations-manual/5/authentication-authorization/property-based-access-control)

File: modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc [hunk 3]
```diff
@@ -97,4 +97,33 @@ The role `regularUsers` does not need to have `READ` privilege for the property
 [source, syntax, role="noheader"]
 ----
 DENY READ {*} ON GRAPH * FOR (n) WHERE NOT n.classification IN ['UNCLASSIFIED', 'PUBLIC'] TO regularUsers
-----
\ No newline at end of file
+----
+
+.Granting permission to `READ` all properties on nodes where the property `createdAt` is later than current date to role `regularUsers`:
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc:102]:
```suggestion
.Granting permission to `READ` all properties on nodes where the property `createdAt` is later than the current date to role `regularUsers`:
```

File: modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc [hunk 4]
```diff
@@ -97,4 +97,33 @@ The role `regularUsers` does not need to have `READ` privilege for the property
...
+The `date()` function is evaluated, and the value used to evaluate the privilege is the date when the property-based privilege is created. 
+Keep this in mind when designing your property rules, and use the `SHOW PRIVILEGES AS COMMANDS` command to check the stored value.
+This is essential when revoking property-based privileges containing evaluated function values like `date()`.
+====
+[NOTE]
+====
+Not all temporal values are comparable, see link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/syntax/operators/#cypher-ordering[Cypher Manual -> Syntax -> Operators -> Ordering and comparison of values].
+====
+
+.Show the privilege created by the command in the example above as a revoke command:
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc:118]:
```suggestion
.Show the privilege created by the command in the previous example as a revoke command:
```

**Review by renetapopova — COMMENTED** [review]
Thanks, @HannesSandberg. I have two minor suggestions. For which version is this PR?

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1887.surge.sh

Updated pages:
- [Property-based access control](https://neo4j-docs-operations-1887.surge.sh/operations-manual/5/authentication-authorization/property-based-access-control)

**Comment by HannesSandberg** [comment]
> For which version is this PR?

@renetapopova Thanks for reviewing. This feature was merged last week so will be for the next release, which would be 5.26 right?

**Comment by renetapopova** [comment]
> > For which version is this PR?
> 
> @renetapopova Thanks for reviewing. This feature was merged last week so will be for the next release, which would be 5.26 right?

yes, thanks!

**Review by renetapopova — APPROVED** [review]

File: modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc [hunk 5]
```diff
@@ -99,16 +99,31 @@ The role `regularUsers` does not need to have `READ` privilege for the property
...
 ====
 Not all temporal values are comparable, see link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/syntax/operators/#cypher-ordering[Cypher Manual -> Syntax -> Operators -> Ordering and comparison of values].
-====
\ No newline at end of file
+====
+
+.Show the privilege created by the command in the previous example as a revoke command:
+[source, syntax, role="noheader"]
+----
+SHOW ROLE regularUsers PRIVILEGES AS REVOKE COMMANDS
```
[the remark this answers is not in the document]
Comment (HannesSandberg) [reply modules/ROOT/pages/authentication-authorization/property-based-access-control.adoc:121]: yes, I did want to show the explicit case and not everything that was granted to the role

**Review by Hunterness — APPROVED** [review]
